### PR TITLE
Feature replace pkg loader

### LINKED ISSUE #23069: Module Loader API rewrite
**Description:**
$Title

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by pramodya1994**
Fixes #23069

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/CompilerDriver.java [hunk 1]
```diff
@@ -193,26 +193,26 @@ void loadLangModules(List<PackageID> pkgIdList) {
...
         if (langLib.equals(QUERY)) {
             // Query module requires stream, array, map, string, table, xml & value modules. Hence loading them.
-            symbolTable.langArrayModuleSymbol = pkgLoader.loadPackageSymbol(ARRAY, null, null);
-            symbolTable.langMapModuleSymbol = pkgLoader.loadPackageSymbol(MAP, null, null);
-            symbolTable.langStringModuleSymbol = pkgLoader.loadPackageSymbol(STRING, null, null);
-            symbolTable.langValueModuleSymbol = pkgLoader.loadPackageSymbol(VALUE, null, null);
-            symbolTable.langXmlModuleSymbol = pkgLoader.loadPackageSymbol(XML, null, null);
-            symbolTable.langTableModuleSymbol = pkgLoader.loadPackageSymbol(TABLE, null, null);
-            symbolTable.langStreamModuleSymbol = pkgLoader.loadPackageSymbol(STREAM, null, null);
+            symbolTable.langArrayModuleSymbol = pkgLoader.loadPackageSymbol(ARRAY, null);
```
Comment (hevayo) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/CompilerDriver.java:200]: Lets have an overloaded function without passing null in each case.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/module/resolver/BaloCache.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package org.wso2.ballerinalang.compiler.packaging.module.resolver;
```
Comment (anupama-pathirage) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/module/resolver/BaloCache.java:1]: Missing license header

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/module/resolver/BirCache.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package org.wso2.ballerinalang.compiler.packaging.module.resolver;
```
Comment (anupama-pathirage) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/module/resolver/BirCache.java:1]: Missing license header

**Comment by anupama-pathirage** [comment]
Some commit messages and the PR title does not match to our guidelines. Refer [1] for more details.

A properly formed Git commit subject line/ PR title should always be able to complete the following sentence:

If applied, this commit/PR will your `subject line here`


[1] https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md

**Comment by hevayo** [comment]
Closing since the design has changed

**Pull request closed without merging**